config/samples/config.yaml

```diff
@@ -0,0 +1,26 @@
+clusterName: "example-cluster"
+timeIntervalSec: 60
+repoCredentials:
+  "https://github.com/rustyTest/testprivateRepo":
```

Comment:
This kind of configuration looks weird to me. 

1. Was it because the k8s `secretKeySelecotr` can not be used to directly with viper, we end up with writing our own?
2. `SSHCredential`, `SSHKey`... do not follow the same pattern as others (lower case first letter).
3. A key like `"git@github.com:rustyTest/testprivateRepo.git"` is tricky, what is the reason we do not user an array?

Reply:
> 3\. A key like `"git@github.com:rustyTest/testprivateRepo.git"` is tricky, what is the reason we do not use an array?
It should be:
```
repoCredentials: 
   "github.com/rustyTest/testprivateRepo":
        SSHCredential:
          ...
        HTTPCredential:
          ....
```

As far as array, I made the assumption that a user should not need more than one ssh credential or http credential. Do you disagree?

Reply:
What I meant was to make the value of `repoCredentials` an array, so that we do not need to use a key like `github.com/rustyTest/testprivateRepo`, instead, to use something like `url: github.com/rustyTest/testprivateRepo`.

Reply:
> What I meant was to make the value of repoCredentials an array, so that we do not need to use a key like github.com/rustyTest/testprivateRepo, instead, to use something like url: github.com/rustyTest/testprivateRepo.

Yeah, that makes sense.

Reply:
sure